coordinator/schedulepod/main.go

```diff
@@ -0,0 +1,694 @@
+	logger := zap.NewExample().Sugar()
+	client, err := metadata.NewClient(addr, logger)
+	if err != nil {
+		return nil, err
+	}
+	etcdConnect := fmt.Sprintf("%s:%s", etcdHost, etcdPort)
+	cli, err := clientv3.New(clientv3.Config{Endpoints: []string{etcdConnect}})
+	if err != nil {
+		return nil, err
+	}
+	kubeJobSpawner := coordinator.KubernetesJobSpawner{}
+	return coordinator.NewCoordinator(client, logger, cli, &kubeJobSpawner)
+}
+
+func testScheduleTrainingSet() error {
+	if err := runner.RegisterFactory(string(runner.CREATE_TRAINING_SET), runner.TrainingSetRunnerFactory); err != nil {
+		return fmt.Errorf("Failed to register training set runner factory: %v", err)
+	}
+	defer runner.UnregisterFactory(string(runner.CREATE_TRAINING_SET))
+	logger := zap.NewExample().Sugar()
+	client, err := metadata.NewClient(metadata_addr, logger)
+	if err != nil {
+		return fmt.Errorf("Failed to connect: %v", err)
+	}
+	defer client.Close()
+	etcdConnect := fmt.Sprintf("%s:%s", etcdHost, etcdPort)
+	cli, err := clientv3.New(clientv3.Config{Endpoints: []string{etcdConnect}})
+	if err != nil {
+		return err
+	}
+	defer cli.Close()
+	featureName := createSafeUUID()
+	labelName := createSafeUUID()
+	tsName := createSafeUUID()
+	serialPGConfig := postgresConfig.Serialize()
+	my_provider, err := provider.Get(provider.PostgresOffline, serialPGConfig)
+	if err != nil {
+		return fmt.Errorf("could not get provider: %v", err)
+	}
+	my_offline, err := my_provider.AsOfflineStore()
+	if err != nil {
+		return fmt.Errorf("could not get provider as offline store: %v", err)
+	}
+	offline_feature := provider.ResourceID{Name: featureName, Variant: "", Type: provider.Feature}
+	schemaInt := provider.TableSchema{
+		Columns: []provider.TableColumn{
+			{Name: "entity", ValueType: provider.String},
+			{Name: "value", ValueType: provider.Int},
+			{Name: "ts", ValueType: provider.Timestamp},
+		},
+	}
+	featureTable, err := my_offline.CreateResourceTable(offline_feature, schemaInt)
+	if err != nil {
+		return fmt.Errorf("could not create feature table: %v", err)
+	}
+	for _, value := range testOfflineTableValues {
+		if err := featureTable.Write(value); err != nil {
+			return fmt.Errorf("could not write to offline feature table")
+		}
+	}
+	offline_label := provider.ResourceID{Name: labelName, Variant: "", Type: provider.Label}
+	labelTable, err := my_offline.CreateResourceTable(offline_label, schemaInt)
+	if err != nil {
+		return fmt.Errorf("could not create label table: %v", err)
+	}
+	for _, value := range testOfflineTableValues {
+		if err := labelTable.Write(value); err != nil {
+			return fmt.Errorf("could not write to offline label table")
+		}
+	}
+	originalTableName := createSafeUUID()
+	if err := CreateOriginalPostgresTable(originalTableName); err != nil {
+		return err
+	}
+	sourceName := createSafeUUID()
+	if err := createTrainingSetWithProvider(client, serialPGConfig, sourceName, featureName, labelName, tsName, originalTableName, "*/1 * * * *"); err != nil {
```

Comment:
Doesnt seem like the table is being updated anywhere?

Reply:
table is updated when a create resource request is sent with a schedule in the config. The coordinator, seeing the schedule is not null, will create an update job seperate to the original create job and push that as a cron job